### PR TITLE
docs(caching): add Partitioning section to Using Pre-aggregations page

### PR DESCRIPTION
**Pull request by hassankhan**
Story details: https://app.clubhouse.io/cube-dev/story/246

### CODE REVIEW

File: docs/content/Caching/Using-Pre-Aggregations.md [hunk 1]
```diff
@@ -123,21 +123,46 @@ strategy has certain limitations:
...
+incremental refreshes; when configured, Cube.js will only refresh partitions as
+necessary. Without incremental refreshing, Cube.js will re-calculate the entire
+pre-aggregation whenever [the refresh key][ref-preaggs-refresh-key] changes.
+
 ## Pre-Aggregations Storage
 
-When using **external** pre-aggregations, Cube.js will
-store pre-aggregations inside its own purpose-built storage layer: Cube Store.
+When using **external** pre-aggregations, Cube.js will store pre-aggregations
+inside its own purpose-built storage layer: Cube Store.
```
Comment (rpaik) [inline docs/content/Caching/Using-Pre-Aggregations.md:147]: add reference to the Cube Store doc?

Comment (hassankhan) [reply docs/content/Caching/Using-Pre-Aggregations.md:147]: We don't have any Cube Store docs available right now - coming soon 👀

File: docs/content/Caching/Using-Pre-Aggregations.md [hunk 2]
```diff
@@ -123,21 +123,46 @@ strategy has certain limitations:
 We highly recommend leaving `readOnly` unset or explicitly setting it to
 `false`.
 
+## Partitioning
+
+[Partitioning][wiki-partitioning] is an extremely effective optimization for
+improving data access. It effectively "shards" the data between multiple tables,
+splitting them by a defined attribute. An incoming query would be checked for
+this attribute, and any valid partitions required to satisfy it are selected.
```
Comment (igorlukanin) [inline docs/content/Caching/Using-Pre-Aggregations.md:131]: Will it work better if we say that "...unnecessary data won't be scanned and processed" (and that's why the reduced refresh time and all)?

Comment (hassankhan) [reply docs/content/Caching/Using-Pre-Aggregations.md:131]: I've changed it to the following:

```
An incoming query would be checked for
this attribute, and **only** valid partitions required to satisfy it are
selected. This results in faster refresh times due to unnecessary data not being
scanned and processed, and possibly even reduced cost, depending on your
database solution.
```

**Comment by keydunov** [comment]
👍

**Comment by keydunov** [comment]
@hassankhan Looks great. Could you please resolve conflicts and merge?

**Pull request merged**